--- cryptlib-3.3.2/random/unix.c.orig	2009-05-12 09:26:50.541650042 +0200
+++ cryptlib-3.3.2/random/unix.c	2009-05-12 09:36:43.777835448 +0200
@@ -106,7 +106,7 @@
 #if !( defined( __QNX__ ) || defined( __CYGWIN__ ) )
   #include <sys/shm.h>
 #endif /* QNX || Cygwin */
-#if defined( __linux__ ) && ( defined(__i386__) || defined(__x86_64__) )
+#if defined( __linux__ ) && ( defined(__i586__) || defined(__i686__) || defined(__pentium4__) || defined(__athlon__) || defined(__x86_64__) )
   #include <sys/timex.h>	/* For rdtsc() */
 #endif /* Linux on x86 */
 #include <signal.h>
